### PR TITLE
feat: add this_model var to model statements

### PR DESCRIPTION
**Pull request by tobymao**
additionally only cache queries with no arguments since it's getting complicated caching everything

### CODE REVIEW

File: sqlmesh/core/renderer.py [hunk 1]
```diff
@@ -61,11 +59,22 @@ def __init__(
         self._default_catalog = default_catalog
         self._quote_identifiers = quote_identifiers
         self.update_schema({} if schema is None else schema)
-        self._cache: t.Dict[CacheKey, t.List[t.Optional[exp.Expression]]] = {}
+        self._cache: t.List[t.Optional[exp.Expression]] = []
+        self._model_fqn = model_fqn
```
Comment (izeigerman) [inline sqlmesh/core/renderer.py:63]: I don't see this being passed into the `ExpressionRenderer` anywhere? Shouldn't we do this while rendering pre-/post- statements?

File: sqlmesh/core/renderer.py [hunk 2]
```diff
@@ -61,11 +59,22 @@ def __init__(
...
 
     def update_schema(self, schema: t.Dict[str, t.Any]) -> None:
         self.schema = d.normalize_mapping_schema(schema, dialect=self._dialect)
 
+    def _should_cache(self, runtime_stage: RuntimeStage, *args: t.Any) -> bool:
+        return runtime_stage == RuntimeStage.LOADING and not any(args)
+
+    def _to_table_mapping(
```
Comment (izeigerman) [inline sqlmesh/core/renderer.py:71]: [Nit] Can we move private methods to the bottom of the class def

File: sqlmesh/core/renderer.py [hunk 3]
```diff
@@ -61,11 +59,22 @@ def __init__(
...
+        snapshots: t.Iterable[Snapshot], deployability_index: t.Optional[DeployabilityIndex]
+    ) -> t.Dict[str, str]:
+        from sqlmesh.core.snapshot import to_table_mapping
+
+        return to_table_mapping(snapshots, deployability_index)
```
Comment (izeigerman) [inline sqlmesh/core/renderer.py:76]: Why not just call this directly?

Comment (tobymao) [reply sqlmesh/core/renderer.py:76]: because i’d have to import teicr

File: sqlmesh/core/renderer.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/renderer.py:63]: done

**Review by izeigerman — APPROVED** [review]

**Pull request merged**